pkg/profilecreator/profilecreator.go

```diff
@@ -312,6 +307,34 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 
 }
 
+// getCPUsSequentially returns Reserved and Isolated CPUs sequentially
+func (ghwHandler GHWHandler) getCPUsSequentially(reservedCPUCount int, topologyInfoNodes []*topology.Node) (string, string, error) {
+	totalCPUSet := cpuset.NewBuilder()
```

Comment:
this path (and PPC in general) isn't a hot path, plus the fist iteration should aim for clarity rather than raw speed so I'd go a step or two down the path of clarity and add another private helper function
```
func totalCPUSetFromTopology(topologyInfoNodes []*topology.Node) cpuset.CPUSet {
    totalCPUSet := cpuset.NewBuilder()
	for _, node := range topologyInfoNodes {
		for _, processorCores := range node.Cores {
			for _, core := range processorCores.LogicalProcessors {
				totalCPUSet.Add(core)
			}
		}
	}
    return totalCPUSet.Result()
}
```
And use it in both helpers. So the explicit loops in the helpers only concern about allocating reserved cores maximizing clarity.

Reply:
Okay, if you think that maximizes clarity, we can certainly do that.